## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 425879`
Commit: `d77cb20193b2e03b3efcee8d0c283d95eb2b3317`

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.pandera</li>
    <li>python312Packages.pandera.dist</li>
    <li>python313Packages.pandera</li>
    <li>python313Packages.pandera.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 32 packages built:</summary>
  <ul>
    <li>python312Packages.layoutparser</li>
    <li>python312Packages.layoutparser.dist</li>
    <li>python312Packages.mlcroissant</li>
    <li>python312Packages.mlcroissant.dist</li>
    <li>python312Packages.pandantic</li>
    <li>python312Packages.pandantic.dist</li>
    <li>python312Packages.pandas-stubs</li>
    <li>python312Packages.pandas-stubs.dist</li>
    <li>python312Packages.pdfplumber</li>
    <li>python312Packages.pdfplumber.dist</li>
    <li>python312Packages.pyreadstat</li>
    <li>python312Packages.pyreadstat.dist</li>
    <li>python312Packages.unstructured-inference</li>
    <li>python312Packages.unstructured-inference.dist</li>
    <li>python313Packages.layoutparser</li>
    <li>python313Packages.layoutparser.dist</li>
    <li>python313Packages.mlcroissant</li>
    <li>python313Packages.mlcroissant.dist</li>
    <li>python313Packages.pandantic</li>
    <li>python313Packages.pandantic.dist</li>
    <li>python313Packages.pandas-stubs</li>
    <li>python313Packages.pandas-stubs.dist</li>
    <li>python313Packages.pdfplumber</li>
    <li>python313Packages.pdfplumber.dist</li>
    <li>python313Packages.pyreadstat</li>
    <li>python313Packages.pyreadstat.dist</li>
    <li>python313Packages.unstructured-inference</li>
    <li>python313Packages.unstructured-inference.dist</li>
    <li>readstat</li>
    <li>tocpdf</li>
    <li>tocpdf.dist</li>
    <li>unstructured-api</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.pandera</summary>
<pre>
tests/strategies/test_strategies.py: 24 warnings
  /private/tmp/nix-build-python3.12-pandera-0.24.0.drv-0/source/tests/strategies/test_strategies.py:900: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
    return pandas_obj.applymap(lambda x: min_val <= x <= max_val)

tests/strategies/test_strategies.py: 30 warnings
  /private/tmp/nix-build-python3.12-pandera-0.24.0.drv-0/source/pandera/strategies/pandas_strategies.py:1021: UserWarning: Dataframe check doesn't have a defined strategy. Falling back to filtering drawn values based on the check definition. This can considerably slow down data-generation.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/polars/test_polars_dtypes.py::test_polars_nested_dtypes_try_coercion[coercible_dtype3-noncoercible_dtype3-data3] - assert False
 +  where False = equals(shape: (1, 1)\n┌─────────────────┐\n│ 0               │\n│ ---             │\n│ list[str]       │\n╞═════════════════╡\n│ ["a", "b", "c"] │\n└─────────────────┘)
 +    where equals = shape: (0, 1)\n┌───────────┐\n│ 0         │\n│ ---       │\n│ list[str] │\n╞═══════════╡\n└───────────┘.equals
 +      where shape: (0, 1)\n┌───────────┐\n│ 0         │\n│ ---       │\n│ list[str] │\n╞═══════════╡\n└───────────┘ = select(<Expr ['col("0")'] at 0x3628791F0>)
 +        where select = shape: (0, 2)\n┌───────────┬──────────────┐\n│ 0         ┆ check_output │\n│ ---       ┆ ---          │\n│ list[str] ┆ bool         │\n╞═══════════╪══════════════╡\n└───────────┴──────────────┘.select
 +          where shape: (0, 2)\n┌───────────┬──────────────┐\n│ 0         ┆ check_output │\n│ ---       ┆ ---          │\n│ list[str] ┆ bool         │\n╞═══════════╪══════════════╡\n└───────────┴──────────────┘ = ParserError("Could not coerce  LazyFrame with schema Schema({'0': List(String)}) into type List(Int64)").failure_cases
 +    and   shape: (1, 1)\n┌─────────────────┐\n│ 0               │\n│ ---             │\n│ list[str]       │\n╞═════════════════╡\n│ ["a", "b", "c"] │\n└─────────────────┘ = collect()
 +      where collect = <LazyFrame at 0x3499403E0>.collect
= 1 failed, 3391 passed, 47 skipped, 12 deselected, 8 xfailed, 7629 warnings in 120.78s (0:02:00) =</pre>
</details>
<details>
<summary>python313Packages.pandera</summary>
<pre>
tests/strategies/test_strategies.py: 24 warnings
  /private/tmp/nix-build-python3.13-pandera-0.24.0.drv-0/source/tests/strategies/test_strategies.py:900: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
    return pandas_obj.applymap(lambda x: min_val <= x <= max_val)

tests/strategies/test_strategies.py: 30 warnings
  /private/tmp/nix-build-python3.13-pandera-0.24.0.drv-0/source/pandera/strategies/pandas_strategies.py:1021: UserWarning: Dataframe check doesn't have a defined strategy. Falling back to filtering drawn values based on the check definition. This can considerably slow down data-generation.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/polars/test_polars_dtypes.py::test_polars_nested_dtypes_try_coercion[coercible_dtype3-noncoercible_dtype3-data3] - assert False
 +  where False = equals(shape: (1, 1)\n┌─────────────────┐\n│ 0               │\n│ ---             │\n│ list[str]       │\n╞═════════════════╡\n│ ["a", "b", "c"] │\n└─────────────────┘)
 +    where equals = shape: (0, 1)\n┌───────────┐\n│ 0         │\n│ ---       │\n│ list[str] │\n╞═══════════╡\n└───────────┘.equals
 +      where shape: (0, 1)\n┌───────────┐\n│ 0         │\n│ ---       │\n│ list[str] │\n╞═══════════╡\n└───────────┘ = select(<Expr ['col("0")'] at 0x16FDFF950>)
 +        where select = shape: (0, 2)\n┌───────────┬──────────────┐\n│ 0         ┆ check_output │\n│ ---       ┆ ---          │\n│ list[str] ┆ bool         │\n╞═══════════╪══════════════╡\n└───────────┴──────────────┘.select
 +          where shape: (0, 2)\n┌───────────┬──────────────┐\n│ 0         ┆ check_output │\n│ ---       ┆ ---          │\n│ list[str] ┆ bool         │\n╞═══════════╪══════════════╡\n└───────────┴──────────────┘ = ParserError("Could not coerce  LazyFrame with schema Schema({'0': List(String)}) into type List(Int64)").failure_cases
 +    and   shape: (1, 1)\n┌─────────────────┐\n│ 0               │\n│ ---             │\n│ list[str]       │\n╞═════════════════╡\n│ ["a", "b", "c"] │\n└─────────────────┘ = collect()
 +      where collect = <LazyFrame at 0x34FCF1C70>.collect
= 1 failed, 3386 passed, 47 skipped, 17 deselected, 8 xfailed, 7629 warnings in 118.60s (0:01:58) =</pre>
</details>
</details>

--- a/pkgs/by-name/re/readstat/package.nix
+++ b/pkgs/by-name/re/readstat/package.nix
@@ -5,7 +5,7 @@
   fetchpatch,
   autoreconfHook,
   pkg-config,
-  libiconv,
+  libtool,
 }:
 
 stdenv.mkDerivation rec {
@@ -20,6 +20,14 @@ stdenv.mkDerivation rec {
   };
 
   patches = [
+    # Remove `gettext` requirement
+    # https://github.com/WizardMac/ReadStat/issues/341
+    (fetchpatch {
+      url = "https://github.com/WizardMac/ReadStat/pull/342/commits/b5512b32d3b3c39e2f0c322df1339a3c61f73712.patch";
+      hash = "sha256-k1yeplrx3pFPl5qzLfsAaj+qunv1BqOZypA05xSolaQ=";
+    })
+
+    # Add (void) to remove -Wstrict-prototypes warnings
     (fetchpatch {
       url = "https://github.com/WizardMac/ReadStat/commit/211c342a1cfe46fb7fb984730dd7a29ff4752f35.patch";
       hash = "sha256-nkaEgusylVu7NtzSzBklBuOnqO9qJPovf0qn9tTE6ls=";
@@ -36,9 +44,8 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     pkg-config
     autoreconfHook
+    libtool
   ];
-
-  buildInputs = [ libiconv ];
 
   enableParallelBuilding = true;
 

